docs/library/forms/form.md

```diff
@@ -0,0 +1,527 @@
+---
+components:
+    - rx.radix.primitives.form.FormRoot
+    - rx.radix.primitives.form.FormField
+    - rx.radix.primitives.form.FormControl
+    - rx.radix.primitives.form.FormLabel
+    - rx.radix.primitives.form.FormMessage
+    - rx.radix.primitives.form.FormSubmit
+---
+
+# Form
+
+```python exec
+import reflex as rx
+import reflex.components.radix.themes as rdxt
+import reflex.components.radix.primitives as rdxp
+```
+
+Forms are used to collect information from your users. Forms group the inputs and submit them together.
+
+This implementation is based on the [Radix forms](https://www.radix-ui.com/primitives/docs/components/form).
+
+## Basic Example
+
+Here is an example of a form collecting an email address, with built-in validation on the email. If email entered is invalid, the form cannot be submitted. Note that the `form_submit` button is not automatically disabled. It is still clickable, but does not submit the form data. After successful submission, an alert window shows up and the form is cleared. There are a few `flex` containers used in the example to control the layout of the form components.
+
+```python demo
+rdxp.form_root(
+    rdxp.form_field(
+        rdxt.flex(
+            rdxp.form_label("Email"),
+            rdxp.form_control(
+                rdxt.textfield_input(
+                    placeholder="Email Address",
+                    # type attribute is required for "typeMismatch" validation
+                    type="email",
+                ),
+                as_child=True,
+            ),
+            rdxp.form_message("Please enter a valid email", match="typeMismatch"),
+            rdxp.form_submit(
+                  rdxt.button("Submit"),
+                  as_child=True,
+            ),
+            direction="column",
+            gap="2",
+            align="stretch",
+        ),
+        name="email",
+    ),
+    on_submit=lambda _: rx.window_alert("Submitted!"),
```

Comment:
We should window alert the form submission: `lambda content: rx.window_alert(content.to_string())`